src/ChunkDef.h

```diff
@@ -85,6 +85,50 @@ class cChunkDef
 	/** The storage wrapper used for compressed nibbledata residing in RAMz */
 	typedef std::vector<NIBBLETYPE> COMPRESSED_NIBBLETYPE;
 
+    static bool IsMoreThanChunkHeight(int v)
+    {
+        return v >= Height;
+    }
+
+    static bool IsLessThanChunkHeight(int v)
+    {
+        return v < 0;
+    }
+
+    static bool IsValidHeight(int v)
+    {
+        return !IsMoreThanChunkHeight(v) && !IsLessThanChunkHeight(v);
+    }
+
+    static bool IsMoreThanChunkWidth(int v)
+    {
+        return v >= Width;
+    }
+
+    static bool IsLessThanChunkWidth(int v)
+    {
+        return v < 0;
+    }
+
+    static bool IsValidWidth(int v)
+    {
+        return !IsMoreThanChunkWidth(v) && !IsLessThanChunkWidth(v);
+    }
+
+    static bool IsValidCoordinates(int x, int y, int z)
+    {
+        return IsValidWidth(x) && IsValidHeight(y) && IsValidWidth(z);
+    }
+
+    static void AssertCoordinatesIsValid(int x, int y, int z)
```

Comment:
We use the `a_` prefix for all function parameters. Also, the coords are expected relative, so I'd suggest `a_RelX` and so on.
Similarly, the function name should probably reflect the fact that it operates on relative coords, so the `Rel` keyword should be included in the name. How about `AssertRelCoordsAreValid()`?

Similarly for the other functions.